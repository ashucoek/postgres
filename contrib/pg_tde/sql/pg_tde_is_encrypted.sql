\set tde_am tde_heap
\i sql/pg_tde_is_encrypted.inc