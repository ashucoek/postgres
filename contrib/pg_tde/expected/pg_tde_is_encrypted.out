CREATE EXTENSION IF NOT EXISTS pg_tde;
SELECT pg_tde_add_database_key_provider_file('file-vault','/tmp/pg_tde_test_keyring.per');
 pg_tde_add_database_key_provider_file 
---------------------------------------
 
(1 row)

SELECT pg_tde_set_key_using_database_key_provider('test-db-key','file-vault');
 pg_tde_set_key_using_database_key_provider 
--------------------------------------------
 
(1 row)

CREATE TABLE test_enc (
  id SERIAL,
  PRIMARY KEY (id)
) USING tde_heap;
CREATE TABLE test_norm (
  id SERIAL,
  PRIMARY KEY (id)
) USING heap;
CREATE TABLE test_part (
  id SERIAL,
  PRIMARY KEY (id)
) PARTITION BY RANGE (id) USING tde_heap;
CREATE TEMP TABLE test_temp_enc (
  id SERIAL,
  PRIMARY KEY (id)
) USING tde_heap;
CREATE TEMP TABLE test_temp_norm (
  id SERIAL,
  PRIMARY KEY (id)
) USING heap;
SELECT relname, amname FROM pg_class JOIN pg_am ON pg_am.oid = pg_class.relam WHERE relname IN ('test_enc', 'test_norm', 'test_part', 'test_temp_enc', 'test_temp_norm') ORDER BY relname;
    relname     |  amname  
----------------+----------
 test_enc       | tde_heap
 test_norm      | heap
 test_part      | tde_heap
 test_temp_enc  | tde_heap
 test_temp_norm | heap
(5 rows)

SELECT relname, pg_tde_is_encrypted(relname) FROM (VALUES ('test_enc'), ('test_norm'), ('test_part'), ('test_temp_enc'), ('test_temp_norm')) AS r (relname) ORDER BY relname;
    relname     | pg_tde_is_encrypted 
----------------+---------------------
 test_enc       | t
 test_norm      | f
 test_part      | 
 test_temp_enc  | t
 test_temp_norm | f
(5 rows)

SELECT relname, pg_tde_is_encrypted(relname) FROM (VALUES ('test_enc_id_seq'), ('test_norm_id_seq'), ('test_part_id_seq'), ('test_temp_enc'), ('test_temp_norm')) AS r (relname) ORDER BY relname;
     relname      | pg_tde_is_encrypted 
------------------+---------------------
 test_enc_id_seq  | t
 test_norm_id_seq | f
 test_part_id_seq | t
 test_temp_enc    | t
 test_temp_norm   | f
(5 rows)

SELECT relname, pg_tde_is_encrypted(relname) FROM (VALUES ('test_enc_pkey'), ('test_norm_pkey'), ('test_part_pkey'), ('test_temp_enc'), ('test_temp_norm')) AS r (relname) ORDER BY relname;
    relname     | pg_tde_is_encrypted 
----------------+---------------------
 test_enc_pkey  | t
 test_norm_pkey | f
 test_part_pkey | 
 test_temp_enc  | t
 test_temp_norm | f
(5 rows)

SELECT pg_tde_is_encrypted(NULL);
 pg_tde_is_encrypted 
---------------------
 
(1 row)

SELECT key_provider_id, key_provider_name, key_name
    FROM pg_tde_key_info();
 key_provider_id | key_provider_name |  key_name   
-----------------+-------------------+-------------
               1 | file-vault        | test-db-key
(1 row)

DROP TABLE test_temp_norm;
DROP TABLE test_temp_enc;
DROP TABLE test_part;
DROP TABLE test_norm;
DROP TABLE test_enc;
DROP EXTENSION pg_tde;
